packages/front-end/components/Queries/AsyncQueriesModal.tsx

```diff
@@ -20,6 +20,7 @@ const AsyncQueriesModal: FC<{
 
   const [showStats, setShowStats] = useState(false);
   const hasStats = data?.queries?.some((q) => q.statistics !== undefined);
+  const datasourceId = data?.queries?.find((q) => q.datasource)?.datasource;
```

Comment:
My understanding from reading through the codepaths is that any grouped queries showing up in this modal will all be on the same datasource, so it's fine to just grab the first one. Does that sound right?

Reply:
Yes, everywhere this modal is currently used it pulls from one datasource. I could imagine us changing that in the future but probably not worth worrying about right now.